src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -16,71 +16,104 @@
 
 package com.google.finapp;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import static com.google.common.truth.Truth.assertThat;
 
 import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.Database;
 import com.google.cloud.spanner.DatabaseClient;
-import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.IntegrationTest;
+import com.google.cloud.spanner.IntegrationTestEnv;
 import com.google.cloud.spanner.Key;
 import com.google.cloud.spanner.KeySet;
 import com.google.cloud.spanner.ResultSet;
-import com.google.cloud.spanner.Spanner;
-import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.testing.RemoteSpannerHelper;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
 import java.math.BigDecimal;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
 import java.util.UUID;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
 
+@Category(IntegrationTest.class)
 public class FinAppIT {
 
-  private static SpannerDaoInterface JDBCDao;
-  private static SpannerDaoInterface JavaDao;
+  private static SpannerDaoInterface spannerDao;
   private static DatabaseClient databaseClient;
+  private static Database db;
 
-  static {
-    // set all IDs according to emulator setup
-    String spannerProjectId = "test-project";
-    String spannerInstanceId = "test-instance";
-    String spannerDatabaseId = "test-database";
-    JDBCDao = new SpannerDaoJDBCImpl(spannerProjectId, spannerInstanceId, spannerDatabaseId);
-    SpannerOptions spannerOptions = SpannerOptions.getDefaultInstance();
-    Spanner spanner = spannerOptions.toBuilder().build().getService();
-    databaseClient = spanner
-        .getDatabaseClient(DatabaseId.of(spannerProjectId, spannerInstanceId, spannerDatabaseId));
-    JavaDao = new SpannerDaoImpl(databaseClient);
-  }
+  @ClassRule
+  public static IntegrationTestEnv env = new IntegrationTestEnv();
 
-  private SpannerDaoInterface getImpl(String impl) {
-    if (impl.equals("java client")) {
-      return JavaDao;
+  @BeforeClass
+  public static void setup() throws IOException {
+    RemoteSpannerHelper testHelper = env.getTestHelper();
+    db = testHelper
+        .createTestDatabase(
+            extractStatementsFromSDLFile("src/main/java/com/google/finapp/schema.sdl"));
+    databaseClient = testHelper.getDatabaseClient(db);
+    if (System.getProperty("SPANNER_USE_JDBC") == null || System.getProperty("SPANNER_USE_JDBC")
+        .equalsIgnoreCase("false")) {
+      spannerDao = new SpannerDaoImpl(databaseClient);
     } else {
```

Comment:
This means that any input other than "false" will be considered as "true". I'm not sure if it's worth it to specifically make sure that the input is "true" and throw an error or something if not.